scripts/setup.sh

```diff
@@ -26,27 +26,37 @@ validate_internet
 f_echo "Advancing with the setup requires the following:"
 f_echo "- Agreeing to Xilinx'/AMD's EULAs (which can be obtained by extracting the installation binary)"
 f_echo "- Enabling WebTalk data collection for version 2021.1 and agreeing to corresponding terms"
-f_echo "- Installation of Rosetta 2 and agreeing to Apple's corresponding software license agreement"
-f_echo "Proceed [Y/n]?"
-read rosetta_install
-case $rosetta_install in
-[yY]|[yY][eE]*)
-	f_echo "Installing Rosetta..."
-	if ! softwareupdate --install-rosetta --agree-to-license
-	then
-		f_echo "Error installing Rosetta."
-		exit 1
-	fi
-	;;
-[nN]|[nN][oO]*)
-	f_echo "Aborting setup."
-	exit 1
-	;;
-*)
-	f_echo "Invalid option."
-	exit 1
-	;;
-esac
+# Check if the Mac is Intel or Apple Silicon
+if [[ "$(uname -m)" == "x86_64" ]]; then
+    f_echo "Mac is Intel-based. Rosetta installation is not required."
+else
+	f_echo "- Installation of Rosetta 2 and agreeing to Apple's corresponding software license agreement"
+    if /usr/bin/pgrep oahd >/dev/null 2>&1; then
```

Comment:
This method of checking whether Rosetta is installed or not [is inaccurate.](https://apple.stackexchange.com/a/464165) Change this and I'll aprove your PR.

Reply:
Updated